File: QTD2.Domain/Entities/Core/Procedure_SaftyHazard_Link.cs
namespace QTD2.Domain.Entities.Core
{
    public class Procedure_SaftyHazard_Link : Common.Entity
    {
        public int ProcedureId { get; set; }

        public int SaftyHazardId { get; set; }

        public virtual Procedure Procedure { get; set; }

        public virtual SaftyHazard SaftyHazard { get; set; }

        public Procedure_SaftyHazard_Link(Procedure procedure, SaftyHazard saftyHazard)
        {
            ProcedureId = procedure.Id;
            SaftyHazardId = saftyHazard.Id;
            Procedure = procedure;
            SaftyHazard = saftyHazard;
        }

        public Procedure_SaftyHazard_Link()
        {
        }
    }
}

File: QTD.SPA/src/app/_DtoModels/Procedure_SaftyHazard_Link/Procedure_SaftyHazard_Link.ts
import { Procedure } from '../Procedure/Procedure';
import { SaftyHazard } from '../SaftyHazard/SaftyHazard';

export class Procedure_SaftyHazard_Link {
  procedureId!: any;
  saftyHazardIds: any[] = [];
  procedure!: Procedure;
  saftyHazard!: SaftyHazard;
  changeNotes!:string;
  effectiveDate?:Date;
}
